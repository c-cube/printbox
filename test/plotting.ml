module B = PrintBox
module BPlot = PrintBox_ext_plot

(* This tests the printbox-ext-plot extension. *)

let test ~size =
  B.extension
  @@ BPlot.(
       Plot
         {
           default_config with
           size;
           specs =
             [
               Scatterplot
                 { points = [| 0., 1.; 1., 0.; 0.75, 0.75 |]; pixel = "#" };
             ];
         })

let () =
  print_endline "Text output:";
  print_endline @@ PrintBox_text.to_string
  @@ test ~size:BPlot.default_config.size;
  print_endline "\nHTML output:";
  print_endline @@ PrintBox_html.to_string @@ test ~size:(800, 800)
