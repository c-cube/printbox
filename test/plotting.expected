Text output:
┌──┬────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ 1│#                                                                                                   │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                          #                         │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│y │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│  │                                                                                                    │
│ 0│                                                                                                   #│
├──┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  │0                                                                                                  1│
│  │                                                 x                                                  │
└──┴────────────────────────────────────────────────────────────────────────────────────────────────────┘

HTML output:
<div><div style="border:thin solid"><table class="framed"><tr><td><table class="non-framed"><tr><td><div>y</div></td><td><table class="non-framed"><tr><td><div>1</div></td></tr><tr><td><div>0</div></td></tr></table></td></tr></table></td><td><div style="border:thin dotted;position:relative;width:800;height:800"><span style="position:absolute;top:0px;left:0px">#</span><span style="position:absolute;top:200px;left:599px">#</span><span style="position:absolute;top:799px;left:799px">#</span></div></td></tr><tr><td><div></div></td><td><table class="non-framed"><tr><td><table class="non-framed"><tr><td><div>0</div></td><td><div class="align-right"><div>1</div></div></td></tr></table></td></tr><tr><td><div class="center"><div>x</div></div></td></tr></table></td></tr></table></div></div>

